x-pack/plugins/threat_intelligence/public/modules/cases/utils/attachments.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CaseAttachmentsWithoutOwner } from '@kbn/cases-plugin/public';
+import { CommentType, ExternalReferenceStorageType } from '@kbn/cases-plugin/common';
+import { JsonValue } from '@kbn/utility-types';
+import { ExternalReferenceAttachmentType } from '@kbn/cases-plugin/public/client/attachment_framework/types';
+import React from 'react';
+import { EuiAvatar } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EMPTY_VALUE } from '../../../common/constants';
+import { Indicator, RawIndicatorFieldId } from '../../../../common/types/indicator';
+import { getIndicatorFieldAndValue } from '../../indicators';
+
+const ExternalAttachmentTypeId = 'indicator';
+
+/**
+ * Indicator name, type, feed name and first seen values,
+ * rendered in the comment section of a case's attachment or in the flyout
+ */
+export interface AttachmentMetadata {
+  indicatorName: string;
+  indicatorType: string;
+  indicatorFeedName: string;
+  indicatorFirstSeen: string;
+}
+
+/**
+ * Create an {@link ExternalReferenceAttachmentType} object used to register an external reference
+ * to the case plugin with our Threat Intelligence plugin initializes.
+ *
+ * See documentation here https://docs.elastic.dev/reops/attachment-framework-external-reference
+ *
+ * This object drives for example:
+ * - the icon displayed in the case attachment
+ * - the component that renders the comment in teh case attachment
+ */
+export const generateAttachmentType = (): ExternalReferenceAttachmentType => ({
+  id: ExternalAttachmentTypeId,
+  displayName: 'indicator',
+  getAttachmentViewObject: () => ({
+    event: (
+      <FormattedMessage
+        id="xpack.threatIntelligence.cases.eventDescription"
+        defaultMessage="added an indicator of compromise"
+      />
+    ),
+    timelineAvatar: <EuiAvatar name="indicator" color="subdued" iconType="crosshairs" />,
+    children: React.lazy(async () => {
```

Comment:
As you don't pass any arguments to `initComponent` I think it is better if you do

```
const AttachmentContentLazy = lazy(() => import('../components/attachment_children/attachment_children'));

export const generateAttachmentType = (): ExternalReferenceAttachmentType => ({
...
children: AttachmentContentLazy
});
```

to avoid unnecessary rerenders.

Reply:
a lot cleaner yes, will fix that up thank you! :)